lute/templates/index.html

```diff
@@ -33,6 +33,9 @@
   <p>To get started using Lute, first <a href="/language/list_predefined">load a predefined language and sample text</a>, or <a href="/language/new">create your language.</a>
 </p>
 {% else %}
+  <script>
+    fetch("/refresh_stale_stats", { "cache": "no-store" });
+  </script>
```

Comment:
This _should_ work reasonably well, but couldn't it run into timing issues, resulting in it still being stale?  The fetch will run causing the stats recalc to start, but then the include will run immediately afterwards and the query might run on data that's still being updated.

The flask `include` that follows calls another endpoint ... I don't think there's an easy way to ensure that this call runs after the `fetch` is completed, but I could be wrong.

Reply:
Yeah, I was thinking the same thing. It could _technically_ happen. But I think it would be exceedingly unlikely since the fetch should initiate before any of the ajax requests triggered by the Datatable rendered from the following `include`. Worst case, I think the Datatable will just get `null` for the book stat that hasn't be refreshed yet. Which is why the status bar was rendering before.

Reply:
There's probably a more robust way of doing it, but I can't think of it right now.

Reply:
This feels like an incremental improvement, and even if it's not 100% perfect it's better than the current state.  I'm going to check one thing, we might be able to get the best result by including your fix in the table listing page itself.

Reply:
`templates/book/tablelisting.html` has the following:

```
  $(document).ready(function () {
    ...
    setup_book_datatable("{{ initial_search or '' }}");
    ...
```

If we move that fetch to above the "setup" method, and have `setup_book_datatable` get called on success, then it should resolve any timing issues.  But that could be another issue, as your fix here seems to be a good initial step.  Sounds good, @cblanken ?

Reply:
I can go ahead an fix it up. No need to make it a separate issue.
Were you thinking like this? There isn't really any error checking on the `refresh_stale_stats` endpoint. It just executes the refresh and returns.

```js
  $(document).ready(function () {
    const current_language_id = {{ current_language_id }};
    // Language filter is set up first, because its value
    // is used during the initial datatable draw.

    setup_language_filter(current_language_id);

    fetch("/refresh_stale_stats", { "cache": "no-store" })
    .then(() => {
      setup_book_datatable("{{ initial_search or '' }}");
      move_datatables_controls_to_config_widget();

      $("#datatables_config_toggle").click(function() {
        $("#datatables_config_items").toggle();
      });
    });
  });
```

That should force the execution to be synchronous and it seems to work fine as far as I can tell.

Reply:
Yep just like that. :-)